### DISCUSSION TITLE
Why firebase notification badge count not exceed 9 ?

### DISCUSSION THREAD

**Discussion by erhan355** [invertase/react-native-firebase#4186] [Q&A]
Hi all,
When I am sending push notifications on android(haven't tried ios yet) app badge count is incremented untill it reaches to 9.When I send 10th notification it becomes 10 and in the blink of an eye it drops to 9 and never exceed to 9 apart from how many more notifications I send.It seems to me it is a limit of android os.Any opinion ?

react-native-firebase/messaging version 7.5.0
react-native-firebase 7.3.1

**Comment by mikehardy** [comment]
I wasn't aware badging was even supported on either react-native-firebase >= 6 or android.

It was my understanding that Android it is vendor-specific, and specifically not implemented on AOSP/Pixel devices, it is not a part of the platform and was only possible via hacks (https://github.com/leolin310148/ShortcutBadger) that haven't been updated in years because people have essentially given up, with the vendor-specificness and platform fighting them

**Comment by erhan355** [comment]
Yes indeed it shows the  count of unread messages on pie 9.In the first place I thought that I could use firebase.notifications.getBadge() and SetBadge() methods but realized that you removed those functionalities on latest major releases. I think using notifee library solves that problem.Am I right ?

**Comment by mikehardy** [comment]
Perhaps, I'd check the APIs of the 3 packages we've had success reports for - notifee, react-native-notifications and react-native-push-notification to see if they really do cover the feature you need.
